src/jenesis/contracts/deploy.py

```diff
@@ -46,6 +46,14 @@ def insert_address(contract_address_names: List[str], deployment: Deployment, pr
 
     return init_data
 
+def retreive_init_addresses(data: Union[Dict, List], contract_names: list, init_addresses: list):
```

Comment:
Minor point, but if you know the content of the list you can use `List`
```suggestion
def retreive_init_addresses(data: Union[Dict, List], contract_names: List[str], init_addresses: List[str]):
```
Mostly I find that these type hints help for future additions / edits / debugging.